client.go

```diff
@@ -163,25 +167,35 @@ func (c Client) findIDInConfigurationsPayload(body io.ReadCloser, name string) (
 	return "", err
 }
 
+// ListNodes returns a list of dataset agent nodes from Flocker Control Service
+func (c *Client) ListNodes() (nodes []NodeState, err error) {
+	resp, err := c.get(c.getURL("state/nodes"))
+	if err != nil {
+		return
```

Comment:
> https://tour.golang.org/basics/7

Well yeah, but 
 * it doesn't follow the house style....that is, I don't see "naked return" statements anywhere else in this library, and 
 * I (at least) find it really hard to read. And I'm not the only one:
   * https://github.com/kubernetes/kubernetes/pull/4263#discussion_r24597352
   * https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters